src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java

```diff
@@ -55,6 +61,140 @@
+        TOKENS_IN_CONFIG_TO_IGNORE.put("EmptyBlock",
+                Stream.of("LITERAL_CATCH", "ARRAY_INIT", "LITERAL_DEFAULT", "LITERAL_CASE")
+                        .collect(Collectors.toSet()));
+        // too many false javadocs missing on override methods and
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("WriteTag",
+                Stream.of("ENUM_CONSTANT_DEF", "METHOD_DEF", "CTOR_DEF", "ANNOTATION_FIELD_DEF")
+                        .collect(Collectors.toSet()));
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("AnnotationLocation",
+                Stream.of("TYPECAST", "DOT", "TYPE_ARGUMENT", "ANNOTATION_DEF", "LITERAL_NEW",
+                        "LITERAL_THROWS", "PARAMETER_DEF", "IMPLEMENTS_CLAUSE",
+                        "ANNOTATION_FIELD_DEF").collect(Collectors.toSet()));
+        // not normally possible to keep these on 1 line due to line length restriction
+        TOKENS_IN_CONFIG_TO_IGNORE.put("NoLineWrap",
+                Stream.of("METHOD_DEF", "CTOR_DEF", "CLASS_DEF", "ENUM_DEF", "INTERFACE_DEF")
+                        .collect(Collectors.toSet()));
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("RightCurly",
+                Stream.of("METHOD_DEF", "CTOR_DEF", "CLASS_DEF", "INSTANCE_INIT", "LITERAL_FOR",
+                        "STATIC_INIT", "LITERAL_WHILE", "LITERAL_DO").collect(Collectors.toSet()));
+        // whitespace after is preferred
+        TOKENS_IN_CONFIG_TO_IGNORE.put("NoWhitespaceAfter",
+                Stream.of("TYPECAST", "LITERAL_SYNCHRONIZED").collect(Collectors.toSet()));
+        // some need context to decide if should be separated or not, others are
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("SeparatorWrap",
+                Stream.of("RBRACK", "AT", "ELLIPSIS", "SEMI", "LPAREN", "RPAREN",
+                        "ARRAY_DECLARATOR").collect(Collectors.toSet()));
+        // prefer no braces, lambda can be on or off depending on context
+        TOKENS_IN_CONFIG_TO_IGNORE.put("NeedBraces",
+                Stream.of("LITERAL_DEFAULT", "LITERAL_CASE", "LAMBDA").collect(Collectors.toSet()));
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("FinalParameters",
+                Stream.of("FOR_EACH_CLAUSE", "LITERAL_CATCH").collect(Collectors.toSet()));
+        // we prefer no spaces on one side or both for these tokens
+        TOKENS_IN_CONFIG_TO_IGNORE.put("WhitespaceAround",
+                Stream.of("WILDCARD_TYPE", "ARRAY_INIT", "GENERIC_END", "GENERIC_START").collect(
+                        Collectors.toSet()));
+        // state of the configuration when test was made
+        TOKENS_IN_CONFIG_TO_IGNORE.put("OperatorWrap",
+                Stream.of("METHOD_REF").collect(Collectors.toSet()));
```

Comment:
METHOD_REF by google style is on "nl", I think it is good to be on new line too for us.

We need to make issue to extend javadoc for http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#METHOD_REF